src/gems/simulation/output_values.py

```diff
@@ -1,194 +1,145 @@
+from gems.simulation.output_values_base import BaseOutputValue
 from gems.study.data import TimeScenarioIndex
 
 
 @dataclass
-class OutputValues:
+class OutputVariable(BaseOutputValue):  # <-- RENAMED from Variable
     """
-    Contents variables output values after solver work completion.
+    Contains a single solver variable's values and status.
+    All shared logic is now in BaseOutputValue.
     """
 
-    @dataclass
-    class Variable:
-        """
-        'constant_var':      c1,
-        'time_only_var':     [[t1, t2, t3]],
-        'scenario_only_var': [s1, s2],
-        'time_scenario_var': [[t1s1, t2s1, t3s1], [t1s2, t2s2, t3s2]]
-
-        Internally, the _value attribute will be a dict mapping TimeScenarioIndex(t,s) to float
-        """
+    _basis_status: Dict[TimeScenarioIndex, str] = field(
+        init=False, default_factory=dict
+    )
 
-        _name: str
-        _value: Dict[TimeScenarioIndex, float] = field(init=False, default_factory=dict)
-        _size: Tuple[int, int] = field(init=False, default=(0, 0))
-        _basis_status: Dict[TimeScenarioIndex, str] = field(
-            init=False, default_factory=dict
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, OutputVariable):  # Reference updated
+            return NotImplemented
+        # Check base equality first (name, size, value)
+        if not super().__eq__(other):
+            return False
+        # Then check the unique field
+        return (self.ignore or other.ignore) or (
+            self._basis_status == other._basis_status
         )
-        ignore: bool = field(default=False, init=False)
-
-        def __eq__(self, other: object) -> bool:
-            if not isinstance(other, OutputValues.Variable):
-                return NotImplemented
-            return (self.ignore or other.ignore) or (
-                self._name == other._name
-                and self._size == other._size
-                and self._value == other._value
-            )
 
-        def is_close(
-            self,
-            other: "OutputValues.Variable",
-            *,
-            rel_tol: float = 1.0e-9,
-            abs_tol: float = 0.0,
-        ) -> bool:
-            # From the docs in https://docs.python.org/3/library/math.html#math.isclose
-            # math.isclose(a, b) returns abs(a-b) <= max(rel_tol * max(abs(a), abs(b)), abs_tol)
-            return (self.ignore or other.ignore) or (
-                self._name == other._name
-                and self._size == other._size
-                and self._value.keys() == other._value.keys()
-                and all(
-                    math.isclose(
-                        self._value[key],
-                        other._value[key],
-                        rel_tol=rel_tol,
-                        abs_tol=abs_tol,
-                    )
-                    for key in self._value
-                )
-            )
+    def _set(
```

Comment:
Why do you need this ? Because it is defined in the parent... Ok this was probably not a good idea to declare it in the parent as the child do not all need that...